lib/gossip/p2p_supervisor.ex

```diff
@@ -7,20 +7,21 @@ defmodule Gossip.P2PSupervisor do
     DynamicSupervisor.start_link(__MODULE__, arg, name: Gossip.P2PSupervisor)
   end
 
-  def start_children(supervisor, num_nodes, topology \\ "full") do
+  def start_children(supervisor, num_nodes, topology \\ "full", algorithm \\ "gossip") do
     Logger.info("Starting children genserver....")
 
     if num_nodes <= 0,
       do: raise(ArgumentError, message: "num_nodes(argument 2) should be greater than 0")
 
     child_pids =
-      for _n <- 1..num_nodes do
-        {:ok, child_pid} = DynamicSupervisor.start_child(supervisor, Gossip.Node)
+      for n <- 1..num_nodes do 
+        {:ok, child_pid} = DynamicSupervisor.start_child(supervisor, {Gossip.Node, [n]})
```

Comment:
Pass in keyword list as `[number_of_nodes: n]` and parse options as `Keyword.get(opts, number_of_nodes)`